grails-core/src/main/groovy/grails/util/GrailsMessageSource.groovy

```diff
@@ -0,0 +1,27 @@
+package grails.util
+
+import org.springframework.context.MessageSource
+
+class GrailsMessageSource {
+    static MessageSource getMessageSource(List<MessageSource> messageSources){
+        if(!messageSources) {
+            return null
+        }
+
+        if(messageSources.size() == 1) {
+            return messageSources.get(0)
+        }
+
+        messageSources.each { messageSource ->
```

Comment:
I don't think using `each` and returning a value from the closure works as expected here. In my mind, `return messageSource` will only return from the closure, not the method. Could you use `find` instead?

Reply:
find was the way to go and resulted in much easier to understand code.  Thanks.